lib/protobuf/field/varint_field.rb

```diff
@@ -64,8 +66,11 @@ def acceptable?(val)
       end
 
       def coerce!(val)
+        fail TypeError, "Expected value of type '#{type_class}' for field #{name}, but got '#{val.class}'" unless acceptable?(val)
         return val.to_i if val.is_a?(Numeric)
         Integer(val, 10)
+      rescue
+        fail TypeError, "Expected value of type '#{type_class}' for field #{name}, but got '#{val.class}'"
```

Comment:
Can we just catch the specific error and reraise? I don't love catch all rescues because it's hard to detect other things breaking.